src/Uno.UI/UI/Xaml/Application.wasm.cs

```diff
@@ -65,6 +67,32 @@ private void Initialize()
 					this.Log().Debug("Launch arguments: " + arguments);
 				}
 				InitializationCompleted();
+
+				if (!string.IsNullOrEmpty(arguments))
+				{
+					NameValueCollection queryValues = null;
+					try
+					{
+						queryValues = HttpUtility.ParseQueryString(arguments);
+					}
+					catch (Exception ex)
+					{
+						this.Log().Error(
+							"Launch arguments could not be parsed as a query string", ex);
+					}
+					if (queryValues != null)
+					{
+						if (queryValues[Uno.Helpers.ProtocolActivation.QueryKey] is
+							string protocolUriString)
+						{
+							protocolUriString = Uri.UnescapeDataString(protocolUriString);
+							var uri = new Uri(protocolUriString);
```

Comment:
Could this whole block (without the condition) be located in the `ProtocolActivation` helper? Also, same thing here, use `Uri.TryCreate`.